packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -308,20 +310,34 @@ export class WorkspaceCommandContribution implements CommandContribution {
      *
      * @param name the simple file name of the file to validate.
      * @param parent the parent directory's file stat.
+     * @param recursive allow file or folder creation using recursive path
      */
-    protected validateFileName(name: string, parent: FileStat): string {
-        if (!validFilename(name)) {
-            return 'Invalid name, try other';
+    protected async validateFileName(name: string, parent: FileStat, recursive: boolean = false): Promise<string> {
+        if (!name) {
+            return '';
         }
-        if (parent.children) {
-            for (const child of parent.children) {
-                if (new URI(child.uri).path.base === name) {
-                    return 'A file with this name already exists.';
-                }
-            }
+        // do not allow recursive rename
+        if (!recursive && !validFilename(name)) {
+            return 'Invalid file or folder name';
+        }
+        // check and validate each sub-paths
+        const names = name.split(/[\\/]/).some(file => !file || !validFilename(file) || /^\s+$/.test(file));
+        if (names || name.startsWith('/') || name.startsWith(' ') || name.endsWith(' ')) {
+            return `The name <strong>${this.trimFileName(name)}</strong> is not a valid file or folder name.`;
```

Comment:
if there is space in the name the error message will display them but error message may be a little bit wired (hard to see spaces). Maybe it will be nice to say to the user specific error message like there are invalid spaces at the beginning or that it shouldn't start with a / instead of saying (not valid)

Reply:
i could separate those and have it like this:
```
        // check and validate each sub-paths
        if (name.split(/[\\/]/).some(file => !file || !validFilename(file) || /^\s+$/.test(file))) {
            return `The name <strong>${this.trimFileName(name)}</strong> is not a valid file or folder name.`;
        } else if (name.startsWith('/')) {
            return 'Absolute paths or names that starts with / are not allowed';
        } else if (name.startsWith(' ') || name.endsWith(' ')) {
            return 'Names with leading or trailing whitespaces are not allowed'
        }
```

Reply:
actually, absolute path check must go first, so itll be like :
```
        if (name.startsWith('/')) {
            return 'Absolute paths or names that starts with / are not allowed';
        } else if (name.startsWith(' ') || name.endsWith(' ')) {
            return 'Names with leading or trailing whitespaces are not allowed';
        }
        // check and validate each sub-paths
        if (name.split(/[\\/]/).some(file => !file || !validFilename(file) || /^\s+$/.test(file))) {
            return `The name <strong>${this.trimFileName(name)}</strong> is not a valid file or folder name.`;
        } 
```